packages/app/src/providers/h5wasm/models.ts

```diff
@@ -0,0 +1,44 @@
+import type { AttributeValues, EntityKind } from '@h5web/shared';
+
+export interface H5WasmEntityResponse {
+  name: string;
+  type:
+    | EntityKind.Dataset
+    | EntityKind.Group
+    | 'external_link'
+    | 'soft_link'
+    | 'other';
+}
+
+export interface H5WasmDatasetReponse extends H5WasmEntityResponse {
+  type: EntityKind.Dataset;
+  dtype: string;
+  shape: number[];
+  attributes: H5WasmAttribute[];
+}
```

Comment:
Sorry - this is part of the cleanup I didn't do before posting.  The H5WasmDatasetResponse is not actually used in the api.  I realized that I didn't need any of the ...Response classes, since the Entity could be directly constructed in  getEntity. (and there is no external service to query for a response).  You could construct "responses" from the h5wasm api calls and then process them, but that seemed to add nothing to the logic since it is pretty simple.